src/master-playlist-controller.js

```diff
@@ -1235,17 +1242,35 @@ export class MasterPlaylistController extends videojs.EventTarget {
   }
 
   /**
-   * Calculates the desired forward buffer length based on current time
+   * Calculates or sets the desired forward buffer length based on current time
    *
-   * @return {Number} Desired forward buffer length in seconds
+   * @param {Number=} Desired forward buffer length in seconds
+   * @return {Number} Current forward buffer target length
    */
-  goalBufferLength() {
-    const currentTime = this.tech_.currentTime();
-    const initial = Config.GOAL_BUFFER_LENGTH;
-    const rate = Config.GOAL_BUFFER_LENGTH_RATE;
-    const max = Math.max(initial, Config.MAX_GOAL_BUFFER_LENGTH);
+  goalBufferLength(value) {
+    if (value === undefined) {
+      const currentTime = this.tech_.currentTime();
+      const initial = this.goalBufferLength_;
+      const rate = Config.GOAL_BUFFER_LENGTH_RATE;
+      const max = Math.max(initial, this.maxGoalBufferLength_);
+
+      return Math.min(initial + currentTime * rate, max);
+    }
+    this.goalBufferLength_ = Math.min(this.goalBufferLength_, value);
+    this.maxGoalBufferLength_ = Math.min(this.maxGoalBufferLength_, value);
+  }
 
-    return Math.min(initial + currentTime * rate, max);
+  /**
+   * Set or return the back buffer length
+   *
+   * @param {Number=} Desired back buffer length in seconds
+   * @return {Number} Current back buffer target length
+   */
+  backBufferLength(value) {
+    if (value === undefined) {
+      return this.backBufferLength_;
+    }
+    this.backBufferLength_ = Math.min(this.backBufferLength_, value);
```

Comment:
Similar feedback as goalBufferLength, we should return the backBufferLength_ value after we set it

Reply:
Correct. Do you think that's a problem? I figure once we hit a limit because the segments are too big, it makes sense to keep that limit in case we're experiencing bandwidth fluctuations since we could jump back up to the rate that caused the decrease in the first place.

Reply:
Even if another part of the code tries to make the buffers larger, the `Math.min` will prevent it. While it's true that you'd expect an actual setter to always set the value, this is not an actual property setter.

If your argument is semantic -- that a thing that looks and sounds like a setter should behave like a setter -- then I understand where you're coming from. They could, I suppose, be split into functions like `getGoalBufferLength` and `safelySetGoalBufferLength` (or better named versions) to break the semantic paradigm, if you think that would avoid any confusion. However, I don't think changing the actual underlying behaviour would be beneficial -- that safety is important for proper functioning.

Reply:
I believe that only `goalBufferLength()` already exists in master and `backBufferLength()` is new in this PR, right? If so, then we could probably rename `backBufferLength()` to be closer to its purpose?